### PR TITLE
Redo convertColors and stripFormatting to be more explicit and support formatting codes

### PR DESCRIPTION
**Pull request by chyyran**
This pull request fixes issue #1. It involved a rewrite of the stripFormatting() and convertColors() method

Also fixed the issue in which kicking a player would not disconnect the player from IRC.

### CODE REVIEW

**Comment by WizardCM** [comment]
Now in build 49.
http://netbook.home.wizardcm.com:8080/job/BukkitIRCd/49/

**Comment by chyyran** [comment]
I'll test this more, though I haven't encountered this issue when I was testing. Perhaps it only works if you use a plugin that enables color codes in chat, not sure if vanilla Bukkit has support for that. It doesn't parse ampersand characters, only section signs, which is used by the ChatColor class to turn ampersands codes to section signs.

**Comment by WizardCM** [comment]
Yeah it's an interesting bug, as before it did send section signs, and now it doesn't.

**Comment by chyyran** [comment]
All it does is replace the section sign plus the code (`§4`) for example, into the corresponding IRC code with the ETX character (`^C05`), `^C` being caret notation for the actual char. Not sure why that bug happens, it doesn't even touch ampersands.

**Comment by WizardCM** [comment]
Well whatever was converting & signs to section signs for them to be parsed for IRC no longer does its job. I'm guessing it was part of the previous converter.

File: src/com/Jdbye/BukkitIRCd/IRCd.java [hunk 1]
```diff
@@ -1530,79 +1529,120 @@ public static boolean writeTo(String nick, String line) {
...
+			output = output.replace(IRC_Ital+"",""); //Get rid of any remaining GS Characters
+			output = output.replace(IRC_Bold+"",""); //Get rid of any remaining STX Characters
+			output = output.replace(IRC_Under+"",""); //Get rid of any remaining US Characters
+
+
+		}else{
+			if (!convertColorCodes){
+				return ChatColor.stripColor(input);
+			}
+			output = input;
```
Comment (chyyran) [inline src/com/Jdbye/BukkitIRCd/IRCd.java:1595]: change to `output = ChatColor.translateAlternateColorCodes('&',input);`

**Comment by chyyran** [comment]
If it's just a problem of converting ampersand codes, I wrote it in such a way that it relied on an external plugin to turn ampersands into section signs, to be non-interferent. If not though, it's an easy fix, just have to add a call to `ChatColors.translateAlternateColorCodes('&',input);`. would turn all ampersands into section signs. Perhaps add a config option to toggle that. See [Line 1595 in IRCd.java](https://github.com/WMCAlliance/BukkitIRCd/pull/29/files#L4R1595)

**Comment by chyyran** [comment]
@WizardCM 

Try [this build](https://buildhive.cloudbees.com/job/ron975/job/BukkitIRCd/14/) and tell me if it's fixed. I couldn't really reproduce the bug, so hopefully that fixes it. If so, I'll make a pull request

**Pull request merged**